plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/frontend.ts

```diff
@@ -70,18 +84,24 @@ const getInputData = (
 	}
 
 	const parsedValue = parseInt( inputElement.value, 10 );
-	const parsedMinValue = parseInt( inputElement.min, 10 );
-	const parsedMaxValue = parseInt( inputElement.max, 10 );
-	const parsedStep = parseInt( inputElement.step, 10 );
-
-	const currentValue = isNaN( parsedValue ) ? 0 : parsedValue;
-	const minValue = isNaN( parsedMinValue ) ? 1 : parsedMinValue;
-	const maxValue = isNaN( parsedMaxValue ) ? undefined : parsedMaxValue;
-	const step = isNaN( parsedStep ) ? 1 : parsedStep;
+	const context = getContext< Context >();
+	const productType = context.productType;
 	const childProductId = parseInt(
 		inputElement.name.match( /\[(\d+)\]/ )?.[ 1 ] ?? '0',
 		10
 	);
+	const id =
+		productType === 'grouped' && childProductId
+			? childProductId
+			: context.productId;
```

Comment:
Here and everywhere else where we have `productType === 'grouped' && childProductId` I wonder if we could simply check for `childProductId`'s existence. `childProductId` will only be defined when the parent product type is `grouped`, so I think it's safe enough, and it would help simplify the code significantly. For example, here it could be:

```suggestion
	const id = childProductId || context.productId;
```

Reply:
Fixed in [1600564](https://github.com/woocommerce/woocommerce/pull/59386/commits/160056417057f0250a6c29cf32e539b5c8abd504)